src/pudl/transform/classes.py

```diff
@@ -851,6 +851,51 @@ def replace_with_na(col: pd.Series, params: ReplaceWithNa) -> pd.Series:
 replace_with_na_multicol = multicol_transform_factory(replace_with_na)
 """A multi-column version of the :func:`nullify_outliers` function."""
 
+################################################################################
+# Spot fix missing values
+################################################################################
+
+
+class SpotFixes(TransformParams):
+    """Parameters that replace certain values with a manual corrected value."""
+
+    record_col: str | None = None
+    """The column used to identify a record."""
+    record_id: str | None = None
```

Comment:
It's worth noting that record id is not always a unique identifier for values. It is for all the plant tables, but not for the others. You can tell whether `record_id` is unique by whether the table transformer class has has the variable`has_unique_record_ids: bool = False` defined. 

This function accommodates that by allowing you to choose any identifying column, but sometimes primary keys are more than one column :/. Because we're just using this for plant tables atm, it's probably ok to keep it as is.

One recommendation would be to include either an `expected_rows_fixed` param or just an assertion in the function that proves that only one row is getting altered based on the information you fed in.

Reply:
Yeah austen is spot on here about the `record_id` not always being a unique identifier. It totally was when I first wrote that issue! So my bad for not giving you to more up-to-date context! I could be nice to generalize this a bit more to take any number of identifying columns + any number of options. I think this would be relatively easy to handle if you set the index as the id columns and used a tuple of ids with `loc`. I had a baby test df in a notebook so I poked at it very quickly to ensure I wasn't suggesting a totally un-fun rabbit hole.

```
from io import StringIO

df = pd.read_csv(
    StringIO(
        """
report_date,plant_id_eia,energy_source_code,prime_mover_code
2019-01-01,8023,DFO,ST
2019-01-01,8023,RC,ST
2019-01-01,8023,SUB,ST
"""
    ),
)

id_cols = ["plant_id_eia", "energy_source_code"]
ids_to_fix = (8023, ["DFO", "SUB"])
df = df.set_index(id_cols)
df.loc[ids_to_fix, "prime_mover_code"] = "AHHH"
df = df.reset_index()
```

Reply:
It looks like non-unique index values are fine based on this little test:

```python
import pandas as pd
import numpy as np

# A few records from our favorite test plant...
barry = pd.DataFrame(
    np.rec.array([(3, '1', '2020-01-01T00:00:00.000000000', 153.1),
           (3, '2', '2020-01-01T00:00:00.000000000', 153.1),
           (3, '3', '2020-01-01T00:00:00.000000000', 272. ),
           (3, '1', '2021-01-01T00:00:00.000000000', 153.1),
           (3, '2', '2021-01-01T00:00:00.000000000', 153.1),
           (3, '3', '2021-01-01T00:00:00.000000000', 272. ),
           (3, '1', '2022-01-01T00:00:00.000000000', 153.1),
           (3, '2', '2022-01-01T00:00:00.000000000', 153.1),
           (3, '3', '2022-01-01T00:00:00.000000000', 272. )],
          dtype=[('plant_id_eia', 'O'), ('generator_id', 'O'), ('report_date', '<M8[ns]'), ('capacity_mw', '<f8')])
)

idx_cols = ["plant_id_eia", "generator_id"]
fix_cols = ["capacity_mw"]
spot_fixes = [
    (3, "1", 1000.0),
    (3, "2", 999.0),
]

spot_fixes_df = pd.DataFrame(spot_fixes, columns=idx_cols+fix_cols).set_index(idx_cols)
barry = barry.set_index(idx_cols)
barry.loc[spot_fixes_df.index, fix_cols] = fixes_df
barry = barry.reset_index()
barry
```

|    |   plant_id_eia |   generator_id | report_date   |   capacity_mw |
|---:|---------------:|---------------:|:--------------|--------------:|
|  0 |              3 |              1 | 2020-01-01    |          1000 |
|  1 |              3 |              2 | 2020-01-01    |           999 |
|  2 |              3 |              3 | 2020-01-01    |           272 |
|  3 |              3 |              1 | 2021-01-01    |          1000 |
|  4 |              3 |              2 | 2021-01-01    |           999 |
|  5 |              3 |              3 | 2021-01-01    |           272 |
|  6 |              3 |              1 | 2022-01-01    |          1000 |
|  7 |              3 |              2 | 2022-01-01    |           999 |
|  8 |              3 |              3 | 2022-01-01    |           272 |

Reply:
Multi-column indexing, as well as options for unique/non-unique index values have been added.